phoenix-core/src/main/java/org/apache/phoenix/coprocessor/DropColumnMutator.java

```diff
@@ -181,13 +183,28 @@ public MetaDataMutationResult validateAndAddMetadata(PTable table,
             byte[] key = mutation.getRow();
             int pkCount = getVarChars(key, rowKeyMetaData);
             if (isView && mutation instanceof Put) {
-                PColumn column = MetaDataUtil.getColumn(pkCount, rowKeyMetaData, table);
+                PColumn column = null;
+                // checking put from the view or index
+                if (Bytes.compareTo(schemaName, rowKeyMetaData[SCHEMA_NAME_INDEX]) == 0
+                        && Bytes.compareTo(tableName, rowKeyMetaData[TABLE_NAME_INDEX]) == 0) {
+                    column = MetaDataUtil.getColumn(pkCount, rowKeyMetaData, table);
+                } else {
+                    for(int i = 0; i < table.getIndexes().size(); i++) {
```

Comment:
Is this required even when the parent is a table (table_type='u') or only needed when it is a view (table_type='v')?

Reply:
Synced offline with @yanxinyi . Missed that we have an `isView` check in the outer if condition. Thanks